lib/dns.js

```diff
@@ -38,6 +37,9 @@ function errnoException(err, syscall, hostname) {
   return ex;
 }
 
+function validIpVersion(isIpResult) {
+  return isIpResult !== 0;
```

Comment:
I'm open to suggestions here - it wasn't clear to me when I read this file what `isIp` returns and I had to read the C++ to realize that it returns 4 for ipv4 6 for ipv6 and 0 for "parsing failed".

The goal was to make the fact that 0 is used for "false" here explicit. I don't mind inlining those checked if you'd like but I think that this helps clarify things - I'm open to other approaches too - what do you think?